example/generate.py

```diff
@@ -0,0 +1,71 @@
+from sys import version_info, path
+from os.path import join, abspath, dirname
+
+path.insert(0, abspath(join(dirname(__file__), "..")))
+
+from novelai_api import NovelAI_API
+from novelai_api.utils import (
+    get_encryption_key,
+    decrypt_user_data,
+    map_meta_to_stories,
+    assign_content_to_story,
+)
+from aiohttp import ClientSession
+
+from logging import Logger
+from asyncio import run
+
+import json
+
+# pip3 install pyyaml
+import yaml
+
+filename = join("credentials", "creds_example.txt")
+with open(filename) as f:
+    username, password = f.read().split(",")
+
+logger = Logger("NovelAI")
+
+
+async def main():
+    async with ClientSession() as session:
+        api = NovelAI_API(session, logger=logger)
+
+        print("login..")
+        login = await api.high_level.login(username, password)
+
+        print("get_encryption_key..")
+        key = get_encryption_key(username, password)
+        keystore = await api.high_level.get_keystore(key)
+        print(keystore)
+
+        print("generate..")
+        input = "sLTgAMYAAgARDt4CIgHeAsYA"
+        model = "genji-python-6b"
+        params = {
+            "generate_until_sentence": True,
+            "max_length": 40,
+            "min_length": 1,
+            "prefix": "vanilla",
+            "num_logprobs": 5,
+            "repetition_pentalty": 1.13125,
+            "repetition_penalty_range": 2048,
+            "repetition_penalty_slope": 0.18,
+            "return_full_text": False,
+            "tail_free_sampling": 1,
+            "temperature": 0.72,
+            "top_k": 0,
+            "top_p": 0.725,
+            "use_cache": False,
+            "use_string": False,
+        }
```

Comment:
A way to load presets would probably be better. It was planned, but there are many things to implement and little time to do them.

Reply:
That's a low level API example though. Presets might be more useful for the high level API.

Reply:
The low level API is not intended for external use, as it is, but a thin wrapper around the raw endpoints. There is no validity check for any of the values in settings as it is high_level’s job to abstract everything, and it is tedious to manage (the same way Keystone was before it got its own type). Generation settings depends on the current model, so it quickly becomes a logistical mess to keep is as raw as that.
I will work on the high_level interface of generate in the next few days, since many things got more granular with Euterpe.

Reply:
Good to know, I'll rescind from making changes here until the high-level generate API is available.

Reply:
The abstraction is out (5b71243). You can see how things are handled in tests/.
It's quite a substantial upgrade, but wasn't too complex (unlike the story management, which is WIP).